src/prefect_server/api/flows.py

```diff
@@ -177,15 +181,22 @@ async def create_flow(
         "version_group_id": {"_eq": version_group_id},
         "tenant_id": {"_eq": tenant_id},
     }
-    # set up a flow group if it's not already in the system
+
+    # lookup the associated flow group (may not exist yet)
     flow_group = await models.FlowGroup.where(
         {
             "_and": [
                 {"tenant_id": {"_eq": tenant_id}},
                 {"name": {"_eq": version_group_id}},
             ]
         }
-    ).first({"id", "schedule"})
+    ).first({"id", "schedule", "settings"})
+
+    # create the flow group or check for the idempotency key in the existing group
+    # Note: The key is stashed in the `settings` dict of the flow group which makes
+    # idempotency only apply to the most flow creation. If we want to avoid
+    # re-registering old flow versions we'll need to add a column to the Flow table
```

Comment:
I think we should remove this comment about old flow versions because it's much more complicated than only adding a single DB column

Reply:
👍 I was thinking that note may be more appropriate in the PR anyway